### PR TITLE
Fix disappearing products in Admin Panel when translation for current locale is missing

### LINKED ISSUE #13296: [AdminBundle] Product isn't displaying if the translation for the current locale is missing
**Sylius version affected**: As far as I know - all versions
**Description**  

If the product has not defined translation for the current locale it isn't displayed on the products list in AdminPanel.

Let's say we have 2 active locales in our shop en_US and pl_PL. We have defined translations for the both locales for the all products. We want to add de_DE locale. We add de_DE locale in AdminPanel, hire the translator and set his AP language to Germany (cause he's German :D). He goes `/admin/products` and sees nothing. The product list is empty.

**Steps to reproduce**  

The easiest way: 
1. Check your shop locales
2. Change your AP language to anyone not defined in the locales list
3. Go to `/admin/products`

**Possible Solution**  

The solutions is pretty simple. In `Sylius\Bundle\CoreBundle\Doctrine\ORM\ProductRepository::createListQueryBuilder` the translation is joined using `innerJoin`. The result is empty list if product title for the current locale is missing. I've managed to fix this by changing `innerJoin` to `leftJoin`.

The only problem I've found is the empty `Name` column in case of not defined title for the current locale. I'm considering the following solutions:
a) If there's no title for the current locale just display '-' - ugly but works, and still better than no products displayed
b) Add some text (?) informing there's no translation for your locale - a lot of work with translating for all supported languages
c) Remove `Name` column (:D) - a little drastic way, but at the end of the day I think it would be the best solution. In my opinion it's enough for a Shop Admin to have Product's code to identify the right product.

I can prepare the PR fixing this problem (it's almost done btw) but I would like to ask you to share your thoughts what would be the best way to deal with `Name` column in that case.

### PR DESCRIPTION
**Pull request by jakubtobiasz**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #13296
| License         | MIT

When accepted I can open similar PRs for `1.9` and `1.10`.

**Description**
Hi 🙋🏻‍♂️!

Let's say our products has translations for `Polish (Poland)` and `English (United States)` locales. Currently if we switch to `Polish` or `English` or any other "non-supported" locale we can't see any product in Admin Panel.

This PR fixes it by displaying all the products and in case of missing name's translation displays `missing translation` info.
CleanShot 2022-03-03 at 7 53 16

Cases covered by tests:
- Products with missing translations sorted by name ascending should be displayed as first
- Products with missing translations sorted by name descending should be displayed as last

These two cases also covers implicitly:
- If product has missing translation display `missing translation` as a product name

In case any change needed just let me know ✌🏻.

### CODE REVIEW

**Review by lchrusciel — APPROVED** [review]
However, I would prefer to have the "allow-plugins" fix separated to another PR. Nonetheless, I'm in favour of both changes

**Comment by lchrusciel** [comment]
As this is bugfx, it should be open to the oldest maintained branch - 1.10

File: src/Sylius/Bundle/UiBundle/Resources/translations/messages.pl.yml [hunk 1]
```diff
@@ -491,6 +491,7 @@ sylius:
         meta_keywords: 'Słowa kluczowe w sekcji meta'
         metadata_index: 'Informacje o dokumencie'
         method: 'metoda'
+        missing_translation: 'brakujące tłumaczenie'
```
Comment (Zales0123) [inline src/Sylius/Bundle/UiBundle/Resources/translations/messages.pl.yml:494]: We should not provide any non-English translation - Crowdin would take care of that 🖖

File: features/product/managing_products/sorting_products.feature [hunk 2]
```diff
@@ -63,3 +64,22 @@ Feature: Sorting listed products
...
+
+    @ui
+    Scenario: Missing translation are sorted as last when sorting by name descending
+        When I change my locale to "Polish"
+        And I browse products
+        And the products are already sorted ascending by name
+        And I sort the products descending by name
+        Then I should see 3 products in the list
+        And the first product on the list should have name "Ekstremalny Mops"
+        And the last product on the list should have name "brakujące tłumaczenie"
```
Comment (Zales0123) [inline features/product/managing_products/sorting_products.feature:85]: We should probably not check the exact text, but rather check is there a test data attribute `data-test-missing-translation` so it would not fail if the translation is changed in Crowdin :)

File: composer.json [hunk 3]
```diff
@@ -240,7 +240,13 @@
...
             "*": "dist"
         },
-        "sort-packages": true
+        "sort-packages": true,
+        "allow-plugins": {
+            "composer/package-versions-deprecated": true,
+            "dealerdirect/phpcodesniffer-composer-installer": true,
+            "symfony/flex": true,
+            "symfony/thanks": true
+        }
```
Comment (Zales0123) [inline composer.json:249]: Should be done in a separate PR

File: src/Sylius/Bundle/UiBundle/Resources/translations/messages.pl.yml (hunk 1 above)
Comment (AdamKasp) [inline src/Sylius/Bundle/UiBundle/Resources/translations/messages.pl.yml:494]: translation different that English should be made crowdin

**Review by Zales0123 — COMMENTED** [review]
Can you open it already to 1.10? It would be upmerged to newer branches 🚀 Thanks for your contribution!

File: features/product/managing_products/sorting_products.feature [hunk 4]
```diff
@@ -63,3 +64,22 @@ Feature: Sorting listed products
         And I sort the products descending by name
         Then I should see 3 products in the list
         And the first product on the list should have name "Szałowy Mops"
+
+    @ui
+    Scenario: Missing translations are sorted as first when sorting by name ascending
+        When I change my locale to "Polish"
+        And I browse products
+        And the products are already sorted ascending by name
```
Comment (AdamKasp) [inline features/product/managing_products/sorting_products.feature:72]:
```suggestion
        And  I sort products  ascending by name
```

File: composer.json (hunk 3 above)
Comment (loic425) [reply composer.json:249]: But I agree to fix it in a separate PR.

**Comment by jakubtobiasz** [comment]
As this PR was based on `master` branch I've created a new PR based on `1.10`. Also I've applied all suggestions from CR. Thanks for all feedback @Zales0123 @lchrusciel @AdamKasp @loic425 

New PR is #13724. I'm closing this PR as it's not needed anymore ✌🏻.
